request_test.go

```diff
@@ -165,6 +167,53 @@ func TestRequestClient(t *testing.T) {
 	req3.chain.assertFailed(t)
 }
 
+func TestRequestWithRetrySuccess(t *testing.T) {
```

Comment:
Can we also check the resulting `retryIndex` in these tests? To ensure that httpexpect made the desired number of retry attempts.